src/components/Header/Header.jsx

```diff
@@ -0,0 +1,64 @@
+import { useState, useEffect } from "react";
+import { useLocation } from "react-router";
+
+import cn from "classnames";
+
+import Btn from "@components/Btn/Btn";
+import BackLink from "./BackLink/BackLink";
+import Logo from "./Logo/Logo";
+import Navigation from "./Navigation/Navigation";
+import Burger from "./Burger/Burger";
+
+import { routes } from "@constants";
+
+import "./Header.css";
+
+const Header = () => {
```

Comment:
u can create some hook like usePageLogic and write all logic there and export it. Then in template u can import that hook and use